cognite_toolkit/_cdf_tk/commands/dump_resource.py

```diff
@@ -90,54 +90,59 @@ def __init__(self, client: ToolkitClient, identifier: DataModelId | None = None)
         self.space_ids: set[str] = set()
 
     def _interactive_select(self) -> DataModelId:
-        include_global = False
-        spaces = self.client.data_modeling.spaces.list(limit=-1, include_global=include_global)
-        selected_space: str = questionary.select(
-            "In which space is your data model located?", [space.space for space in spaces]
-        ).ask()
-
         data_model_ids = self.client.data_modeling.data_models.list(
-            space=selected_space, all_versions=False, limit=-1, include_global=include_global
+            all_versions=False, limit=-1, include_global=False
         ).as_ids()
-
-        if not data_model_ids:
-            raise ToolkitMissingResourceError(f"No data models found in space {selected_space}")
+        available_spaces = sorted({model.space for model in data_model_ids})
+        if not available_spaces:
+            raise ToolkitMissingResourceError("No data models found")
+        if len(available_spaces) == 1:
+            selected_space = available_spaces[0]
+        else:
+            selected_space = questionary.select("In which space is your data model located?", available_spaces).ask()
+        data_model_ids = sorted(
+            [model for model in data_model_ids if model.space == selected_space], key=lambda model: model.as_tuple()
+        )
 
         selected_data_model: DataModelId = questionary.select(
-            "Which data model would you like to dump?", [Choice(f"{model!r}", value=model) for model in data_model_ids]
+            "Which data model would you like to dump?",
+            [
+                Choice(f"{model_id!r}", value=model_id)
+                for model_id in sorted(data_model_ids, key=lambda model: model.as_tuple())
+            ],
         ).ask()
```

Comment:
Above you implicitly select the space when only 1 exists; why do you not implicitly select the model when only one model exists?

Reply:
The method is for dumping a datamodel. Then, I want the user to explicit select one. The reason for first selecting space is to avoid overwhelming the user with all models they have access to. If that is only one, we skip it.